service/storage_service.cc

```diff
@@ -1744,6 +1744,11 @@ class topology_coordinator {
 
     // Returns true if the state machine was transitioned into tablet migration path.
     future<bool> maybe_start_tablet_migration(group0_guard);
+
+    future<> await_event() {
+        _as.check();
```

Comment:
But we have this at the beginning of the coordinator:

```
    auto abort = _as.subscribe([this] () noexcept {
        _topo_sm.event.signal();
    });
```

So in the scenario you describe (if I understand it correctly) the coordinator should be woken.

Reply:
It's not enough, because this signal() can be executed before the coordinator executes await().

Reply:
Right. Although, this implementation details is not documented. Should be document it and rely on it? @avikivity

Reply:
It's not good to rely on this (condition_variable remembering being woken). It's not how standard condition variables work.

Reply:
In general the loop to wait for the condition variable should check any state for which can cause a wakeup.

Reply:
But then we have to emulate this behavior outside of condition variable, like I did with _tablets_ready, or by taking a lock (group0 guard) and atomically releasing it with waiting.

Reply:
If you'd need to do that with a standard condition variable, then that's okay. If we use the name of a standard thing, then it should have the semantics of the standard thing.

Reply:
There is no real condition variable, only documentation.